### PR TITLE
ardupilot: guided commands from RFDesign

### PR DESCRIPTION
**Pull request by tridge**
these are used for fixed wing guided control from companion computer

### CODE REVIEW

File: message_definitions/v1.0/ardupilotmega.xml [hunk 1]
```diff
@@ -241,6 +249,36 @@
         <param index="6">Reserved, send 0</param>
         <param index="7">Reserved, send 0</param>
       </entry>
+      <entry value="43000" name="MAV_CMD_GUIDED_CHANGE_SPEED">
```
Comment (peterbarker) [inline message_definitions/v1.0/ardupilotmega.xml]: A comment here as to when you'd actually use these would be handy.  Or improve the descriptions as "Change target altitude" is the same as (several?) other messages.

File: message_definitions/v1.0/ardupilotmega.xml [hunk 2]
```diff
@@ -241,6 +249,36 @@
...
+        <param index="1">Empty</param>
+        <param index="2">Empty</param>
+        <param index="3">Rate of change (m/s), 0 for maximum rate change</param>
+        <param index="4">Empty</param>
+        <param index="5">Empty</param>
+        <param index="6">Empty</param>
+        <param index="7">Altitude (m)</param>
+      </entry>
+      <entry value="43002" name="MAV_CMD_GUIDED_CHANGE_HEADING">
+        <description>Change to target heading, rather then location flight mode</description>
```
Comment (peterbarker) [inline message_definitions/v1.0/ardupilotmega.xml]: Could we get this translated into English, please?

File: message_definitions/v1.0/ardupilotmega.xml [hunk 3]
```diff
@@ -241,6 +249,36 @@
...
+        <param index="1">Track type (see HEADING_TYPE enum)</param>
```
Comment (peterbarker) [inline message_definitions/v1.0/ardupilotmega.xml]:
```suggestion
        <param index="1">Heading type (see HEADING_TYPE enum)</param>
```

File: message_definitions/v1.0/ardupilotmega.xml [hunk 4]
```diff
@@ -241,6 +249,36 @@
...
+        <description>Change flight speed (affects guided only)</description>
+        <param index="1">Type (see SPEED_TYPE enum)</param>
+        <param index="2">Target Speed (m/s)</param>
+        <param index="3">Acceleration rate (m/s/s), 0 to take effect instantly</param>
+        <param index="4">Empty</param>
+        <param index="5">Empty</param>
+        <param index="6">Empty</param>
+        <param index="7">Empty</param>
+      </entry>
+      <entry value="43001" name="MAV_CMD_GUIDED_CHANGE_ALTITUDE">
```
Comment (peterbarker) [inline message_definitions/v1.0/ardupilotmega.xml]: Why not add a climb rate parameter to `MAV_CMD_DO_CHANGE_ALTITUDE` instead?

File: message_definitions/v1.0/ardupilotmega.xml [hunk 5]
```diff
@@ -241,6 +249,36 @@
...
+        <description>Change target altitude</description>
+        <param index="1">Empty</param>
+        <param index="2">Empty</param>
+        <param index="3">Rate of change (m/s), 0 for maximum rate change</param>
```
Comment (peterbarker) [inline message_definitions/v1.0/ardupilotmega.xml]: Is this the absolute value of the rate of change?

**Comment by davidbuzz** [comment]
I have opened a new PR to answer Peter's concerns here:   https://github.com/ArduPilot/mavlink/pull/109 as I can't push 'fixes'  to the branch Tridge created.

**Pull request closed without merging**